### PR TITLE
MIMXRT595_EVK clock choices for FRO

### PR DESCRIPTION
**Pull request by DerekSnell**
Provides Kconfigs to set some clock muxes to FRO

### CODE REVIEW

**Review by iuliana-prodan — DISMISSED** [review]

File: soc/arm/nxp_imx/rt5xx/Kconfig.soc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # i.MX RT5XX Series
 
-# Copyright (c) 2022, NXP
+# Copyright (c) 2023, NXP
```
[the remark this answers is not in the document]
Comment (DerekSnell) [reply soc/arm/nxp_imx/rt5xx/Kconfig.soc]: Thank you @manuargue and @decsny .  I fixed the copyright

File: soc/arm/nxp_imx/rt5xx/soc.c [hunk 2]
```diff
@@ -458,12 +462,20 @@ void __weak imxrt_pre_init_display_interface(void)
...
+	 */
 	CLOCK_AttachClk(kAUX1_PLL_to_MIPI_DPHY_CLK);
 	CLOCK_InitSysPfd(kCLOCK_Pfd3,
 		((CLOCK_GetSysPllFreq() * 18ull) /
 		((unsigned long long)(DT_PROP(DT_NODELABEL(mipi_dsi), phy_clock)))));
 	CLOCK_SetClkDiv(kCLOCK_DivDphyClk, 1);
-
+#elif defined(CONFIG_MIPI_DPHY_CLK_SRC_FRO)
+	CLOCK_AttachClk(kFRO_DIV1_to_MIPI_DPHY_CLK);
+	CLOCK_SetClkDiv(kCLOCK_DivDphyClk, 1);
```
Comment (danieldegrasse) [inline soc/arm/nxp_imx/rt5xx/soc.c:477]: For the other conditional case, we do some basic calculations to determine the clock divider, based on the `phy-clock` property.

We don't seem to have a function to determine the FRO clock frequency within the SDK, but there is a constant defined: [CLK_FRO_CLK](https://github.com/zephyrproject-rtos/hal_nxp/blob/master/mcux/mcux-sdk/devices/MIMXRT595S/system_MIMXRT595S_cm33.h#L88). Could we do something like the following:

`CLOCK_SetClkDiv(kCLOCK_DivDphyClk, (CLK_FRO_CLK / DT_PROP(DT_NODELABEL(mipi_dsi), phy_clock))));`

I'm not sure if you'd find this valuable, I just want to preserve the functionality of the `phy-clock` property if we can.

File: soc/arm/nxp_imx/rt5xx/soc.c [hunk 3]
```diff
@@ -281,8 +281,12 @@ void __weak rt5xx_clock_init(void)
 	/* Switch SYSTICK_CLK to MAIN_CLK_DIV */
 	CLOCK_AttachClk(kMAIN_CLK_DIV_to_SYSTICK_CLK);
 #if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(flexcomm0), nxp_lpc_usart, okay)
-	/* Switch FLEXCOMM0 to FRG */
-	CLOCK_AttachClk(kFRG_to_FLEXCOMM0);
+	#ifdef CONFIG_FLEXCOMM0_CLK_SRC_FRG
+		/* Switch FLEXCOMM0 to FRG */
+		CLOCK_AttachClk(kFRG_to_FLEXCOMM0);
+	#elif (defined CONFIG_FLEXCOMM0_CLK_SRC_FRO)
```
[the remark this answers is not in the document]
Comment (DerekSnell) [reply soc/arm/nxp_imx/rt5xx/soc.c]: thanks @danieldegrasse , updated as you requested

File: soc/arm/nxp_imx/rt5xx/soc.c (hunk 2 above)
Comment (DerekSnell) [reply soc/arm/nxp_imx/rt5xx/soc.c]: thanks @danieldegrasse , updated as you requested

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**